### PR TITLE
store: fix panic when failed to create pdclient

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #42778 

Problem Summary:

### What is changed and how it works?
- since this pr is more like a extra fix of #42778, so will not add release-note

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
  - create store without real-tikv, check it can fail with err
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: store/driver/tikv_driver.go [hunk 1]
```diff
@@ -188,11 +188,10 @@ func (d TiKVDriver) OpenWithOptions(path string, options ...Option) (resStore kv
 		),
 		pd.WithCustomTimeoutOption(time.Duration(d.pdConfig.PDServerTimeout)*time.Second),
 		pd.WithForwardingOption(config.GetGlobalConfig().EnableForwarding))
-	pdCli = util.InterceptedPDClient{Client: pdCli}
```
Comment (D3Hunter) [inline store/driver/tikv_driver.go:191]: this will cause previous defer close panic when failed to create pdclient

**Review by lance6716 — APPROVED** [review]

**Comment by lance6716** [comment]
@hawkingrei do you know there's a linter that can guard tha if a function returns error, the error must be checked before using other return values?

**Review by hawkingrei — APPROVED** [review]

**Comment by hawkingrei** [comment]
> @hawkingrei do you know there's a linter that can guard tha if a function returns error, the error must be checked before using other return values?

I can't immediately think of it, but I can look for it.

**Pull request merged**